corehq/apps/domain/middleware.py

```diff
@@ -49,13 +47,31 @@ class DomainHistoryMiddleware(MiddlewareMixin):
 
     def process_response(self, request, response):
         if hasattr(request, 'domain') and getattr(response, '_remember_domain', True):
-            self.remember_domain_visit(request, response)
+            self.remember_domain_visit(request)
         return response
 
-    def remember_domain_visit(self, request, response):
-        last_visited_domain = request.session.get('last_visited_domain')
-        if last_visited_domain != request.domain:
-            request.session['last_visited_domain'] = request.domain
+    def remember_domain_visit(self, request):
+        if hasattr(request, 'couch_user') and request.couch_user:
+            last_visited_domain = get_last_visited_domain(request.couch_user)
+            if last_visited_domain != request.domain:
+                set_last_visited_domain(request.couch_user, request.domain)
+
+
+def get_last_visited_domain(couch_user):
+    client = cache_core.get_redis_client()
+    return client.get(_last_visited_domain_cache_key(couch_user))
+
+
+def set_last_visited_domain(couch_user, domain):
+    client = cache_core.get_redis_client()
+    cache_expiration = 60 * 60 * 24
```

Comment:
This seems like a short expiration - it wouldn't persist past a weekend.

Reply:
good point, changed to 7 days in 97c53fb0100